File: backend/src/main/java/com/ecommerce/backend/service/CategoryService.java
package com.ecommerce.backend.service;

import com.ecommerce.backend.dto.request.CreateCategoryRequest;
import com.ecommerce.backend.dto.response.CategoryResponse;

import java.util.List;

public interface CategoryService {

    CategoryResponse createCategory(String tenantName, CreateCategoryRequest request);

    CategoryResponse updateCategory(String tenantName,
                                    Long categoryId,
                                    CreateCategoryRequest request);

    void deleteCategory(String tenantName, Long categoryId);

    List<CategoryResponse> getAllCategories(String tenantName);
}

File: frontend/src/services/categoryService.js
import api from "./api";

export const getCategories = async () => {
  return await api.get(`/categories`);
};

export const createCategory = async (category) => {
  return await api.post(`/categories`, category);
};

export const updateCategory = async (categoryId, category) => {
  return await api.put(`/categories/${categoryId}`, category);
};

export const deleteCategory = async (categoryId) => {
  return await api.delete(`/categories/${categoryId}`);
};